planner/core/task.go

```diff
@@ -1289,6 +1300,82 @@ func BuildFinalModeAggregation(
 	return
 }
 
+// convertAvgForMPP converts avg() to if(count(arg)=0, null, sum(arg)/count(arg)), in detail:
+// 1.rewrite avg() in the final aggregation to count() and sum(), and reconstruct its schema.
+// 2.replace avg() with if(count(arg)=0, null, sum(arg)/count(arg)) and reuse the original schema of the final aggregation.
+func (p *basePhysicalAgg) convertAvgForMPP(prop *property.PhysicalProperty) *PhysicalProjection {
```

Comment:
Why do we need this? Could you give me an example?

Reply:
does `this` mean `needing a prop`?